utils/query_engine/media_wiki.py

```
@@ -133,16 +134,34 @@ def mediawiki_search(query: str) -> dict:
             
-            logger.info(f"Fetching content for: {title}")
-            content = _fetch_page_content(api_url, title)
-            pages_content[title] = content
+            # Collect results as they complete
+            for future in as_completed(future_to_title):
+                title = future_to_title[future]
+                try:
+                    content = future.result()
+                    pages_content[title] = content
+                    logger.info(f"Successfully fetched content for: {title}")
+                except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)